Laravel 5.2 Shift

jasonmccreary opened the pull request:
This pull request includes the changes for upgrading to Laravel 5.2. Feel free to commit any additional changes to the `shift-10241` branch.

**Before merging**, you should:

- Checkout the `shift-10241` branch
- Review **all** pull request comments for additional changes
- Update your dependencies for Laravel 5.2
- Run `composer update` (if the pre-scripts fail, add `--no-scripts`)
- Thoroughly test your application

If you want help with your upgrade, check out the [Shift Human Services](https://laravelshift.com/human-services) or join the [Shifty Coders](https://laravelshift.com/shifty-coders) Slack room to get answers to all your Laravel questions.

laravel-shift commented:
Laravel 5.2 adds *Middleware Groups* for convenience. Since this may require changes in your routes and controllers, I did not upgrade `app/Http/Kernel.php`. You should review [Middleware Groups](http://laravel.com/docs/5.2/middleware#middleware-groups) in more detail. When ready, merge your `app/Http/Kernel.php` with the [5.2 version](https://github.com/laravel/laravel/blob/becd774e049fb451aca0c7dc4f6d86d7bc12256c/app/Http/Kernel.php) and upgrade your routes/controllers accordingly.

laravel-shift commented:
Laravel 5.2 updated the `database` session driver. If you are using the `database` driver, you need to add the `user_id` (nullable integer), `ip_address` (nullable string), and `user_agent` (text) columns to your session database table.

laravel-shift commented:
Laravel 5.2 removed the `pretend` mail configuration option. If you were using `pretend`, use the `log` driver instead.

laravel-shift committed:
Add new exceptions to `$dontReport` property

laravel-shift committed:
Shift Middleware

Laravel 5.2 adjusts the `Guard` object used within middleware. In
addition, new `can` and `throttles` middleware were added.

laravel-shift committed:
Adopt PSR-2 coding style

The Laravel framework adopts the PSR-2 coding style in version 5.1.
Laravel apps *should* adopt this coding style as well. Read the
[PSR-2 coding style guide][1] for more details and check out [PHPCS][2]
to use as a code formatting tool.

[1]: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
[2]: https://github.com/squizlabs/PHP_CodeSniffer

laravel-shift committed:
Shift configuration

laravel-shift committed:
Shift Authentication

Laravel 5.2 introduces a *Guard* object to the Authentication
component. In doing so, the configuration options and default
Middleware changed.

laravel-shift committed:
Shift core files

laravel-shift committed:
Shift Laravel dependencies

jasonmccreary commented:
⚠ Laravel 5.2 adds *Middleware Groups* for convenience. Since this may affect your routes and controllers, Shift did not automate this upgrade. You should read about [Middleware Groups](http://laravel.com/docs/5.2/middleware#middleware-groups) to understand this change.

When you are ready to upgrade to *Middleware Groups*, you should:

- [ ] Compare your `app/Http/Kernel.php` with the [5.2 version](https://github.com/laravel-shift/laravel-5.2/blob/master/app/Http/Kernel.php)
- [ ] Compare your `app/Providers/RouteServiceProvider.php` with the [5.2 version](https://github.com/laravel-shift/laravel-5.2/blob/master/app/Providers/RouteServiceProvider.php)
- [ ] Review any usages of the `web` middleware within your application

jasonmccreary commented:
ℹ Laravel 5.2 changed the value of `$redirectTo` from `/home` to `/` in `app/Http/Controllers/Auth/AuthController.php`. Shift did not make this change since you may have customized it for your app. However, you may wish to change this if you are not using *Auth* or if you want to follow the latest Laravel conventions.

jasonmccreary commented:
ℹ Laravel 5.2 updated the `database` session driver. If you are using the `database` driver, you will need to add the `user_id` (nullable integer), `ip_address` (nullable string), and `user_agent` (text) columns to your session database table.

jasonmccreary commented:
❌ Shift could not upgrade the following configuration files since they differed from the default Laravel version. You will need to compare these configuration files against the [default Laravel 5.2 configuration files](https://github.com/laravel-shift/laravel-5.2/tree/master/config) and merge any changes:

- [ ] config/auth.php

jasonmccreary commented:
ℹ Laravel 5.2 no longer uses the `phpspec/phpspec` package. Unless you require this package for your project, you may remove it from your `composer.json` development dependencies.

jasonmccreary committed:
Default config files for Shift
